src/test/java/org/assertj/core/internal/files/Files_assertHasBinaryContent_Test.java

```diff
@@ -54,53 +55,64 @@ static void setUpOnce() {
 
   @Test
   void should_throw_error_if_expected_is_null() {
-    assertThatNullPointerException().isThrownBy(() -> files.assertHasBinaryContent(someInfo(), actual, null))
+    assertThatNullPointerException().isThrownBy(() -> files.assertHasBinaryContent(INFO, actual, null))
                                     .withMessage("The binary content to compare to should not be null");
   }
 
   @Test
   void should_fail_if_actual_is_null() {
-    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> files.assertHasBinaryContent(someInfo(), null, expected))
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> files.assertHasBinaryContent(INFO, null, expected))
                                                    .withMessage(actualIsNull());
   }
 
   @Test
   void should_fail_if_actual_is_not_file() {
-    AssertionInfo info = someInfo();
+    // GIVEN
     File notAFile = new File("xyz");
-
-    Throwable error = catchThrowable(() -> files.assertHasBinaryContent(info, notAFile, expected));
-
-    assertThat(error).isInstanceOf(AssertionError.class);
-    verify(failures).failure(info, shouldBeFile(notAFile));
+    // WHEN
+    expectAssertionError(() -> files.assertHasBinaryContent(INFO, notAFile, expected));
+    // THEN
+    verify(failures).failure(INFO, shouldBeFile(notAFile));
   }
 
   @Test
   void should_pass_if_file_has_expected_binary_content() throws IOException {
-    when(binaryDiff.diff(actual, expected)).thenReturn(BinaryDiffResult.noDiff());
-    files.assertHasBinaryContent(someInfo(), actual, expected);
+    File actual = newFile(tempDir.getAbsolutePath() + "/tmp.txt");
+    byte[] data = "actual".getBytes();
+    try (FileOutputStream myWriter = new FileOutputStream(actual)) {
+      myWriter.write(data, 0, data.length);
+    }
```

Comment:
this comment applies broadly: use [commons io FileUtils](https://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/FileUtils.html) to write file content, there are many ways, ex `writeByteArrayToFileFile file, byte[] data)` or `write(File file, CharSequence data, String charsetName)`. These methods throw `IOException`, don't catch them just add them in the test method throw declaration.